bdsx/bds/implements.ts

```diff
@@ -612,7 +612,16 @@ ItemStack.prototype.getMaxDamage = procHacker.js("ItemStackBase::getMaxDamage",
 ItemStack.prototype.getDamageValue = procHacker.js("ItemStackBase::getDamageValue", int32_t, {this:ItemStack});
 ItemStack.prototype.isWearableItem = procHacker.js("ItemStackBase::isWearableItem", bool_t, {this:ItemStack});
 ItemStack.prototype.getAttackDamage = procHacker.js("ItemStackBase::getAttackDamage", int32_t, {this:ItemStack});
-ItemStack.prototype.allocateAndSave = procHacker.js("ItemStackBase::save", CompoundTag.ref(), {this:ItemStack});
+const ItemStackBase$save = procHacker.js("ItemStackBase::save", TagPointer, {this:ItemStack}, TagPointer);
+ItemStack.prototype.allocateAndSave = function() {
+    const ptr = new TagPointer(true);
+    ptr.value = CompoundTag.construct();
+    ItemStackBase$save.call(this, ptr);
+    const tag = CompoundTag.construct(ptr.value);
```

Comment:
the returning `Tag` should be allocated by `CompoundTag.allocate()`.
with `allocatedPush` or `allocatedSet` it moved into the `ListTag` or `CompoundTag`.
and it can be deleted by their destructor.

it means it will call `free(tag)`
and it should be allocated with `malloc()`